### PR TITLE
Refactoring of roles and notifications about antagonist assignment

### PR DESCRIPTION
**Pull request by rincew1nd**
## About the PR
- Refactoring of a traitor role assignment code (structuring of code blocks);
- Double traitor role assignment exception prevention;
- Refactoring of Roles with the ability to inherit antagonist roles and extend the functionality of actions on it's assignment;
- Sound files moved to more appropriate path;

**Media**
https://www.youtube.com/watch?v=MeTCW6Wwk3Y

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl: Rincew1nd
- add: All the antag roles have briefing sounds now

### CODE REVIEW

**Review by malchanceux — COMMENTED** [review]
tator alert is not the same as listed in license file

**Comment by ficcialfaint** [comment]
CL has to have a verb, i.e:
❌ add: Cool thing
✅ add: Added cool thing

**Comment by deltanedas** [comment]
also an exception that is very rare shouldnt be in player facing cl

**Comment by rincew1nd** [comment]
@malchanceux audio files were converted to mono, cause positional audio worked only with mono sound files.
The original file was reverted and the method to play audio was changed to global (to prevent further inconvenience).

@ficcialfaint @deltanedas change log was fixed to comply with all requirements.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:49]: Move to PiratesRuleComponent, also see if some gun or something serves as a better temp replacement (maybe a monkey scream?) Also is this even necessary? shouldnt Greet() handle this now?

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Lord i can't wait for the mind refactor so we can purge the role class, but this is good for the fine in the meantime i think

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (deltanedas) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:49]: also it should be a SoundSpecifier not a path to a sound file

File: Content.Server/Roles/ZombieRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/ZombieRole.cs:9]: The reason for this PR is customization, so there is a traitor sound to the zombie role. I haven't looked for a more proper sound for roles, but we have some in a linked PR. I can look into merging it here too.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:49]: Pirates don't have a role, so you can't use the added default functionalities.
I will try to find a replacement sound and will fix a SoundSpecifier.

File: Content.Server/Roles/ZombieRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/ZombieRole.cs:9]: Made "/Audio/Voice/Zombie/zombie-3.ogg" as an alert at the event of the zombie role assignment.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs]: Made "/Audio/Effects/Diseases/monkey1.ogg" as an alert sound to greet new pirates.

**Comment by rincew1nd** [comment]
Additional notification sounds for pirate and zombie role was added.
license.txt was updated to specify a source of audio files.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:253]: no need to use the factory here, take a look at how maketraitor works currently

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:260]: define params in datafield

File: Content.Server/Roles/TraitorRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/TraitorRole.cs:1]: now there is a duplicate Content.Server.Traitor.TraitorRole

File: Content.Server/Roles/AntagonistRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/AntagonistRole.cs:5]: filescoped namespace

File: Content.Server/Roles/NukeopsRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/NukeopsRole.cs:7]: same

File: Content.Server/Roles/TraitorRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/TraitorRole.cs:7]: same

File: Content.Server/Roles/ZombieRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/ZombieRole.cs:7]: same

File: Content.Server/Roles/AntagonistRole.cs
Comment (AJCM-git) [inline Content.Server/Roles/AntagonistRole.cs:48]: i think you should just move the greet sound to whatever antag system/comp that doesn't have something like that yet (i think just zombies) and remove this

File: Resources/Audio/Ambience/Antag/license.txt
Comment (AJCM-git) [inline Resources/Audio/Ambience/Antag/license.txt:1]: convert this to a RGA (there are examples in the codebase, search for files called `attributions.yml`)

File: Resources/Audio/Ambience/Antag/license.txt
Comment (AJCM-git) [inline Resources/Audio/Ambience/Antag/license.txt:5]: are these two already in the game? (iirc you said pirate_start is actually monkey1.ogg so i assume zombie_start is the same) if so please reference those files directly in the code instead of duplicating them, also referencing your own pr in another fork as the source for the sound won't be accepted unless you made it

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/Roles/TraitorRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/TraitorRole.cs:1]: Removed

File: Content.Server/Roles/AntagonistRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/AntagonistRole.cs:5]: Fixed

File: Content.Server/Roles/NukeopsRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/NukeopsRole.cs:7]: Fixed

File: Content.Server/Roles/TraitorRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/TraitorRole.cs:7]: Fixed

File: Content.Server/Roles/ZombieRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/ZombieRole.cs:7]: Fixed

File: Content.Server/Roles/AntagonistRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/AntagonistRole.cs:48]: The purpose of it is to not make a bunch of the same code in different antagonist classes.

File: Resources/Audio/Ambience/Antag/license.txt
Comment (rincew1nd) [reply Resources/Audio/Ambience/Antag/license.txt:5]: Those files are original files from another forked server. We will deal with it today.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:253]: Made it like it's made in the traitor system, without deletion of a comment.

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:241]: Moved it to for-each creation of pirates.

File: Content.Server/Roles/AntagonistRole.cs
Comment (AJCM-git) [reply Content.Server/Roles/AntagonistRole.cs]: playing the greet sound is something so minor that it doesn't warrant adding more code to something that will be removed sooner or later like the role class and its inheritors, just repeat for now

File: Resources/Audio/Ambience/Antag/license.txt
Comment (AJCM-git) [reply Resources/Audio/Ambience/Antag/license.txt]: You should mention who made them then in the RGA (i took a look at the new RGA in this pr and it seems to mention different files?)

Comment (rincew1nd) [reply Resources/Audio/Ambience/Antag/license.txt]: Done, i was waiting for approval of a PR.

File: Resources/Audio/Ambience/Antag/license.txt
Comment (rincew1nd) [reply Resources/Audio/Ambience/Antag/license.txt:1]: Done

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs]: Done

File: Content.Server/Roles/AntagonistRole.cs
Comment (rincew1nd) [reply Content.Server/Roles/AntagonistRole.cs]: Moved to sound notifications to System components.

**Comment by rincew1nd** [comment]
@AJCM-git all comments fixed. PR is ready to be merged.
License files are updated with correct links to sources.
Sound notifications moved from roles to systems.

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:263]: unnecesary argument since PlayGlobal *should* just get the GreetSoundNotification params if none are given

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:263]: same

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:228]: same

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/PiratesRuleSystem.cs:261]: can't you just give it the session without the need of a filter (since IPlayerSession inherits ICommonSession)?

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:36]: Was not aware of this naming style, 'cause in a lot of places the naming varies a lot.

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs:289]: Removed Greet overload from TraitorRole class

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs]: Done

File: Content.Server/GameTicking/Rules/PiratesRuleSystem.cs
Comment (rincew1nd) [reply Content.Server/GameTicking/Rules/PiratesRuleSystem.cs]: This method is explicitly done as an admin option to spawn a pirate.
It will not trigger any logic done at the start of the round at OnPlayerSpawningEvent.
BUT you are right, those methods (SetOutfitCommand.SetOutfit and StationSpawningSystem.EquipStartingGear) do the same stuff, but there are different checks cause of the lack of components at the start of the execution.
I believe that this is a task for refactoring that is out of the scope of this PR.

File: Content.Server/GameTicking/Rules/TraitorRuleSystem.cs
Comment (deltanedas) [reply Content.Server/GameTicking/Rules/TraitorRuleSystem.cs]: huh i thought leaving out System was preferred

File: Resources/Audio/Ambience/Antag/attributions.yml
Comment (metalgearsloth) [inline Resources/Audio/Ambience/Antag/attributions.yml:12]: These need to reference specific commits not a branch.

**Pull request merged**
